metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py

```diff
@@ -118,30 +120,37 @@ def __post_init__(self):
 
 
 class ClickHouseConfig(
-    BasicSQLAlchemyConfig, BaseTimeWindowConfig, DatasetLineageProviderConfigBase
+    TwoTierSQLAlchemyConfig, BaseTimeWindowConfig, DatasetLineageProviderConfigBase
 ):
     # defaults
     host_port = Field(default="localhost:8123", description="ClickHouse host URL.")
     scheme = Field(default="clickhouse", description="", hidden_from_docs=True)
     password: pydantic.SecretStr = Field(
         default=pydantic.SecretStr(""), description="password"
     )
-
-    secure: Optional[bool] = Field(default=None, description="")
-    protocol: Optional[str] = Field(default=None, description="")
-
```

Comment:
I am not quite comfortable with directly removing these as this is breaking change. Can we keep supporting them in recipe - maybe write a pydantic root_validator that adds these values in `uri_opts` config - similar to [this example.](https://github.com/datahub-project/datahub/blob/v0.10.4/metadata-ingestion/src/datahub/ingestion/source/tableau.py#L332)

At the same time, you can also mark these two as deprecated using `pydantic_field_deprecated` . You'll refer the usage in above example.